Recruit-Training/Advanced-Recruit-Training/Telemetry-Teaser/src/App.tsx

```diff
@@ -1,25 +1,118 @@
+import { useState } from "react";
+
 import BatteryInput from "~/components/batteryInput";
 import Header from "~/components/header";
 import SpeedInput from "~/components/speedInput";
 import WeatherInput from "~/components/weatherInput";
 
 const App = () => {
-  const calculateRange = () => {
-    return;
-  }
+  const [speed, setSpeed] = useState<number | undefined>(0);
+  const [speedValid, setSpeedValid] = useState(true);
+  const [battery, setBattery] = useState<number | undefined>(100);
+  const [batteryValid, setBatteryValid] = useState(true);
+  const [weather, setWeather] = useState(50);
+  const [range, setRange] = useState<number | null>(null);
```

Comment:
same thing for the range, do we need to use state for the range? we know what the formula is for the state, so is it possible to directly compute the range via the other state variables we defined?